docs: ephemeral workspaces for workspaces/API (#439)
hashicorp/terraform-docs-common: website/docs/cloud-docs/api-docs/notification-configurations.mdx

| Errored            | `"run:errored"`         | A run terminates early due to error or cancellation.                                          |
| Drifted            | `"assessment:drifted"`  | Terraform Cloud detected configuration drift. This option is only available if you enabled drift detection for the workspace.                                                    |
| Checks Failed      | `"assessment:check_failure"` | One or more continuous validation checks did not pass. This option is only available if you enabled drift detection for the workspace. |

> | Health Assessment Errored | `"assessment:failed`"   | A health assessment failed. This option is only available if you enable health assessments for the workspace.                                        |

Author: Just updated this one to align with the UI implementation, but will remove.

| Auto Destroy Reminder | `"workspace:auto_destroy_reminder`"   | Reminders that a destroy run will be triggered at some point in the future. Reminders are sent 12 and 24 hours before the scheduled auto destroy time. |

> | Auto Destroy Results | `"workspace:auto_destroy_run_results`"   | Final run result for the scheduled auto destroy run. |

Reviewer: 💅 Nit: A subtle note on how these docs differ structurally from the rest. More or less all the other descriptions start by describing the trigger for the notification. These might opt to start with the following, respectively:
`An automated workspace destroy run is imminent.`
`An automated destroy run was attempted.`

Author: great point

}
```

> ### Workspace Notification Payload - Health Assessments

Author: This should be good now, could you verify?

> ### Workspace Notification Payload - Health Assessments

Reviewer 2: Looks good!

### Workspace Notification Payload

Workspace events include detailed information about workspace-level validation events like [health assessments](/terraform/cloud-docs/workspaces/health) if you enable them for the workspace. Much of the information provides details about the associated [assessment result](/terraform/cloud-docs/api-docs/assessment-results), which Terraform Cloud uses to track instances of continuous validation.

> The types of attributes returned in the payload details will vary depending on the type of `trigger_scope`.

Reviewer 2: This feels kinda vague? Do folks need to know what `trigger_scope` is or how it affects payload? We could link somewhere or expand here?
```suggestion
Terraform Cloud returns different types of attributes in the payload details, depending on the type of `trigger_scope`.
```

#### Health Assessments

Health assessment notifications for workspace events contain the following information:

> <!-- BEGIN: TFC:only name:pnp-callout -->

Reviewer: I think this red highlighting and the rest that follows is some sign that syntax/format is off. But, I can't identify what it is.

Reviewer 2: Unfortunately, this syntax highlighting is because we are using a callout and for some reason the HTML comment syntax makes MDX mad. I keep meaning to look this, but haven't gotten a chance yet 😢

| `details.workspace_name`                     | string | Human-readable name of the workspace.                                                |
| `details.organization_name`                  | string | Human-readable name of the organization.                                             |

##### Sample Payload

> Health assessment payloads will have information about resource drift and continuous validation checks.

Reviewer 2: ```suggestion
Health assessment payloads have information about resource drift and continuous validation checks.
```

}
```

### Workspace Notification Payload - Ephemeral Workspaces

<!-- BEGIN: TFC:only name:pnp-callout -->
@include 'tfc-package-callouts/ephemeral-workspaces.mdx'
<!-- END: TFC:only name:pnp-callout -->

| Name                                         | Type   | Description                                                                                         |
| -------------------------------------------- | ------ | ----------------------------------------------------------------------------------------------------|

> | `payload_version`                            | number | Always "2".                                                                                         |

Author: This was copied from the above health assessment payload table. I removed the quotes here, is that what you were thinking?

> | `payload_version`                            | number | Always "2".                                                                                         |

Reviewer 2: Yeah, I think that makes more sense. Thank you for doing that!

| `details.run_created_at`                     | string | Timestamp of when the destroy run was successfully created. (Only applies to results notifications.) |
| `details.run_status`                         | string | Status of the scheduled destroy run. (Only applies to results notifications.)                        |
| `details.run_external_id`                    | string | ID of the scheduled destroy run. (Only applies to results notifications.)                            |
| `details.run_create_error_message`           | string | Message detailing why the run was unable to be queued. (Only applies to results notifications.)      |
| `details.trigger_type`                       | string | The type of notification, values are "reminder" or "results".                                        |
| `details.workspace_name`                     | string | Human-readable name of the workspace.                                                               |
| `details.organization_name`                  | string | Human-readable name of the organization.                                                            |

> #### Sample Payload - Reminder

Author: Could you verify that this looks roughly how you were intending?

> #### Sample Payload - Reminder

Reviewer 2: This looks great! Thank you so much for splitting it up! ✨

}
```

#### Automatic Destroy Runs

<!-- BEGIN: TFC:only name:pnp-callout -->
@include 'tfc-package-callouts/ephemeral-workspaces.mdx'
<!-- END: TFC:only name:pnp-callout -->

> 

Reviewer 2: Can we add a sentence above this table for context. Something like:
```suggestion

Automatic destroy run notifications for workspace events contain the following information:

```

| `details.run_status`                         | string | Status of the scheduled destroy run. Only applies to results notifications.                                  |
| `details.run_external_id`                    | string | The ID of the scheduled destroy run. Only applies to results notifications.                                  |
| `details.run_create_error_message`           | string | Message detailing why the run was unable to be queued. Only applies to results notifications.                |
| `details.trigger_type`                       | string | The type of notification, and the value is either "reminder" or "results".                                   |
| `details.workspace_name`                     | string | Human-readable name of the workspace.                                                                       |
| `details.organization_name`                  | string | Human-readable name of the organization.                                                                    |

##### Sample Payload

> The shape of data in auto destroy notification payloads may differ depending on the success of the created run. Specific cases are called out below.

Reviewer 2: ```suggestion
The shape of data in auto destroy notification payloads may differ depending on the success of the run you created. Refer to the specific examples below.
```

###### Reminder

> Reminders that a destroy run will be triggered at some point in the future.

Reviewer 2: ```suggestion
Reminders that Terraform Cloud will trigger a destroy run at some point in the future.
```

```json
{
  "payload_version": "2",
  "notification_configuration_id": "nc-SZ3V3cLFxK6sqLKn",
  "notification_configuration_url": "https://app.terraform.io/api/v2/notification-configurations/nc-SZ3V3cLFxK6sqLKn",
  "trigger_scope": "workspace",
  "trigger": "workspace:auto_destroy_reminder",
  "message": "Auto Destroy Reminder",
  "details": {
      "auto_destroy_at": "2025-01-01T00:00:00Z",
      "run_created_at": null,
      "run_status": null,
      "run_external_id": null,
      "run_create_error_message": null,
      "trigger_type": "reminder",
      "workspace_name": "learned-english-dog",
      "organization_name": "acme-org"
  }
}
```

###### Results

> The final result of the scheduled auto destroy run will include additional metadata about the run.

Reviewer 2: ```suggestion
The final result of the scheduled auto destroy run includes additional metadata about the run.
```

```json
{
  "payload_version": "2",
  "notification_configuration_id": "nc-SZ3V3cLFxK6sqLKn",
  "notification_configuration_url": "https://app.terraform.io/api/v2/notification-configurations/nc-SZ3V3cLFxK6sqLKn",
  "trigger_scope": "workspace",
  "trigger": "workspace:auto_destroy_results",
  "message": "Auto Destroy Results",
  "details": {
      "auto_destroy_at": null,
      "run_created_at": "2022-06-09T05:22:51Z",
      "run_status": "applied",
      "run_external_id": "run-vRVQxpqq64EA9V5a",
      "run_create_error_message": null,
      "trigger_type": "results",
      "workspace_name": "learned-english-dog",
      "organization_name": "acme-org"
  }
}
```

###### Failed Run Creation

> Run-specific values will be empty when Terraform Cloud was unable to create an auto destroy run.

Reviewer 2: ```suggestion
Run-specific values are empty when Terraform Cloud was unable to create an auto destroy run.
```
